### ISSUE TITLE
Windows Support?

### ISSUE THREAD

**Issue by KyleKing** [eigentechnologies/cpip#1]
I saw this repository linked from https://github.com/sdispater/poetry/issues/190 and thought `cpip` might help resolve an issue on pdoc3 to distribute the package on Anaconda: https://github.com/pdoc3/pdoc/issues

Do you have any interest in expanding `cpip` for Windows? I might be able to help, but I haven't worked with conda outside of installing packages

**Comment by goldslope** [comment]
Regarding Window, unfortunately I have no interest at the moment given a few reasons:

1. I wrote this entirely in Bash, which means Windows would require a complete rewrite
2. Even considering a rewrite, we have to consider the fact that conda does NOT have a publicly documented API for Python, that means if we were to try make it cross-platform by writing in Python, we would be writing on a shaky API foundation
3. cpip leverages Anaconda compiler packages (this is a CRUCIAL component for building PyPi packages in a cross-compatible matter), which are currently not available for Windows.


To help clarify what cpip is for:

cpip is more for creating portable **environments** not **packages**.  For PyPi packages (i.e. via Poetry), it will create binary-compatible wheels when necessary using the Anaconda cross compilers. It will then ship the associated runtime compiler libraries (i.e. libgcc) so you don't run into typical linking errors that occur when compiling wheels on say Ubuntu 16.04, then running them on an older distro like Red Hat 6.

As the issue states, if you actually want to create an Anaconda package for pdoc3, you'll have to follow the instructions for creating conda packages (read [here](https://conda.io/projects/conda-build/en/latest/index.html)).


How/why my company uses it:

At Eigen Technologies, our tech stack consists primarily of Python packages + some thirdparty apps (Mongo, PSQL).  Because we have A LOT of Python packages, and only about half of them are available as conda packages, instead of mixing and matching, we decided to have Poetry manage all of our Python packages (in addition we get robust version locking). Some Python packages are however NOT portable if they are from PyPi, unlike conda packages. This gives us trouble since we sometimes need to install our product and its dependencies offline, sometimes without access to any compilers.

We were thinking about making conda package for every Python package that was not available from Anaconda/conda-forge repositories, but we decided this would be a pain in the ass given how often packages need updating, and how often we add and delete these packages from our stack. instead cpip ensures a portable environment even if all the Python packages are from PyPi.
